### PR TITLE
build: replace deprecated axe dependency

### PR DESCRIPTION
**Pull request by crisbeto**
The `axe-webdriverjs` package was deprecated in favor of `@axe-core/webdriverjs`. These changes switch to the new package, fix the new failures that were picked up and clean up some repetitive code.

### CODE REVIEW

File: src/e2e-app/virtual-scroll/virtual-scroll-e2e.html [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <section class="demo-virtual-scroll-uniform-size">
   <h3>Uniform size</h3>
-  <cdk-virtual-scroll-viewport class="demo-viewport" autosize>
+  <cdk-virtual-scroll-viewport class="demo-viewport" autosize tabindex="0">
```
Comment (jelbourn) [inline src/e2e-app/virtual-scroll/virtual-scroll-e2e.html:3]: Why the `tabindex` here?

Comment (crisbeto) [reply src/e2e-app/virtual-scroll/virtual-scroll-e2e.html:3]: Axe complains that scrollable areas need to have tab stops.

**Review by jelbourn — APPROVED** [review]
LGTM

Huh, the scrollable area thing is not something I've run into before

**Comment by crisbeto** [comment]
For what it's worth, it's what I had tried to tackle in https://github.com/angular/components/pull/14808.

**Comment by jelbourn** [comment]
Right, I didn't think of that; it way indexed in my brain under "tabs".

**Comment by mmalerba** [comment]
bumping to minor due to conflicts on patch branch

**Pull request merged**